amethyst_renderer/src/renderer.rs

```diff
@@ -115,6 +119,16 @@ impl Renderer {
 
         self.device.cleanup();
 
+        // # Enforce maximum FPS
+        // Sleep if the time needed is > 10 ms for 9.5 ms
+        let time_remaining = Duration::from_secs(1) / self.max_fps - start.elapsed();
+        if time_remaining > Duration::new(0, 10000000) {
+            sleep(time_remaining * 95 / 100);
```

Comment:
I'd recommend to wake up for `N` earlier where `2*N` is your treshold for sleeping.
E.g. subtract constant instead of multiplying.